src/platform/packages/shared/kbn-config-schema/src/types/literal_type.ts

```diff
@@ -11,15 +11,26 @@ import { internals } from '../internals';
 import { Type } from './type';
 
 export class LiteralType<T> extends Type<T> {
+  private expectedValue: any;
+
   constructor(value: T) {
     super(internals.any().valid(value));
+    this.expectedValue = value;
   }
 
-  protected handleError(type: string, { value, valids: [expectedValue] }: Record<string, any>) {
+  protected handleError(
+    type: string,
+    { value, valids: [expectedValue = this.expectedValue] = [] }: Record<string, any>
+  ) {
+    const formatter = (v: any) =>
```

Comment:
WDYT about avoid this "valids" array entirely. Something like:

```ts
export class LiteralType<T> extends Type<T> {
  private expectedValue: T;
  constructor(value: T) {
    super(internals.any().valid(value));
    this.expectedValue = value;
  }

  protected handleError(type: string) {
    switch (type) {
      case 'any.required':
      case 'any.only':
        return `expected value to equal [${this.expectedValue}]`;
    }
  }
}
```

Reply:
Also would be nice to remove these `any`s :)

Reply:
Yeah good ideas. Applied all changes in 5a719d9.

I also removed the changes to the error message, partly I wanted to see how many failures it would cause. Nope, not worth it 😅.